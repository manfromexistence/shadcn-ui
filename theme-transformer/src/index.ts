import { otherPresets } from "./presets";

const transformThemeResets = (themeResets) => {
  return Object.entries(themeResets).map(([key, value]) => ({
    name: key,
    label: value.label || key.charAt(0).toUpperCase() + key.slice(1),
    radius: value.radius || "0.5rem",
    fonts: {
      sans: value.fonts?.sans || "default-sans-font",
      serif: value.fonts?.serif || "default-serif-font",
      mono: value.fonts?.mono || "default-mono-font",
    },
    light: value.light,
    dark: value.dark,
  }));
};

// Example theme resets to transform
const themeResets = {
  claude: {
    label: "Claude",
    radius: "0.5rem",
    fonts: {
      sans: "Adamina",
      serif: "Adamina",
      mono: "Fira Code",
    },
    light: {
      background: "#faf9f5",
      foreground: "#3d3929",
      // other light theme properties...
    },
    dark: {
      background: "#262624",
      foreground: "#c3c0b6",
      // other dark theme properties...
    },
  },
  // Add more theme resets as needed...
};

export const transformedThemes = transformThemeResets(themeResets);