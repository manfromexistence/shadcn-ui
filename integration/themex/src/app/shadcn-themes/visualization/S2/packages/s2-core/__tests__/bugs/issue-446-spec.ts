/**
 * @description spec for issue #446
 * https://github.com/antvis/S2/issues/446
 * #asyncGetAllPlainData error in table mode
 *
 */
import { TableSheet } from '@/sheet-type';
import { asyncGetAllPlainData } from '@/utils';
import { TAB_SEPARATOR, type S2Options } from '../../src';
import * as mockDataConfig from '../data/data-issue-446.json';
import { getContainer } from '../util/helpers';

const s2Options: S2Options = {
  width: 800,
  height: 600,
  seriesNumber: {
    enable: true,
  },
};

describe('export', () => {
  test('should export correct data with show seriesNumber', async () => {
    const s2 = new TableSheet(getContainer(), mockDataConfig, s2Options);

    await s2.render();
    const data = await asyncGetAllPlainData({
      sheetInstance: s2,
      split: TAB_SEPARATOR,
      formatOptions: true,
    });

    expect(data.split('\n').length).toEqual(3);
    expect(data.split('\n')[0].split('\t').length).toEqual(4);
    expect(data.split('\n')[0].split('\t')[0]).toEqual('序号');
    expect(data).toMatchSnapshot();
  });

  test('should export correct data without show seriesNumber', async () => {
    const s2 = new TableSheet(getContainer(), mockDataConfig, {
      ...s2Options,
      seriesNumber: {
        enable: false,
      },
    });

    await s2.render();
    const data = await asyncGetAllPlainData({
      sheetInstance: s2,
      split: TAB_SEPARATOR,
    });

    expect(data.split('\n').length).toEqual(3);
    expect(data.split('\n')[0].split('\t').length).toEqual(3);
    expect(data.split('\n')[0].split('\t')[0]).toEqual('col0');
    expect(data).toMatchSnapshot();
  });
});
