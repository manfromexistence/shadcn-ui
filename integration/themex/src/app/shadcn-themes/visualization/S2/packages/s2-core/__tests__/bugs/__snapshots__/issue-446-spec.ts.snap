// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export should export correct data with show seriesNumber 1`] = `
"序号	col0	col1	col2
1	col0-0	col1-0	col2-0
2	col0-1	col1-1	col2-1"
`;

exports[`export should export correct data without show seriesNumber 1`] = `
"col0	col1	col2
col0-0	col1-0	col2-0
col0-1	col1-1	col2-1"
`;
